datasets/openpi/openpi.py

```diff
@@ -0,0 +1,247 @@
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # TODO: This method is tasked with downloading/extracting the data and defining the splits depending on the configuration
+        # If several configurations are possible (listed in BUILDER_CONFIGS), the configuration selected by the user is in self.config.name
+
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        suffix = "_metadata" if self.config.type_ == "metadata" else ""
+        fname = "id_{config}{suffix}.jsonl"
+        urls = {
+            "train": _BASE_URL + "train/" + fname.format(config=self.config.mode, suffix=suffix),
+            "dev": _BASE_URL + "dev/" + fname.format(config=self.config.mode, suffix=suffix),
+            "test": _BASE_URL + "test/" + fname.format(config=self.config.mode, suffix=suffix),
+        }
+        data_dir = dl_manager.download_and_extract(urls)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": data_dir["train"], "split": "train"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": data_dir["test"], "split": "test"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": data_dir["dev"], "split": "dev"},
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        with open(filepath, encoding="utf-8") as f:
+            for id_, row in enumerate(f):
+                data = json.loads(row)
+                if "question" in data:
+                    result = {"id": data["id"], "question": data["question"]}
+                elif "question_metadata" in data:
+                    result = {
+                        "id": data["id"],
+                        "question_metadata": {
+                            "url": data["question_metadata"].get("url", ""),
```

Comment:
Why use get here ? Are some urls fields missing ?
Same for the other fields

If some of them are missing you can keep `get` but maybe you can mention that you replace missing values with empty strings in the dataset card

Reply:
Which section in the card should this go under? I didn't get it from the `README_guide.md`

Reply:
Included in the dataset card